modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationDeploymentTestCase.java

```diff
@@ -1284,24 +1285,50 @@ public void deploysAppWithPluginExportingAlreadyProvidedAppPackage() throws Exce
   }
 
   @Test
-  public void deploysAppWithExportedPackagePrecedenceOverPlugin() throws Exception {
-    // Defines a plugin that contains org.bar package, which is also exported on the application
-    ArtifactPluginFileBuilder echoPluginWithoutLib1 = new ArtifactPluginFileBuilder("echoPlugin1")
-        .configuredWith(EXPORTED_CLASS_PACKAGES_PROPERTY, "org.foo")
-        .containingClass(pluginEcho1TestClassFile, "org/foo/Plugin1Echo.class")
-        .dependingOn(new JarFileBuilder("barUtils2_0", barUtils2_0JarFile));
+  @Issue("MULE-17112")
+  @Description("If a plugin uses a library and the application sets another version of that library as a sharedLib, the plugin internally uses its own version of the lib and not the app's. "
+      + "Similar to deploysAppWithLibDifferentThanPlugin, but the bar2 dep in the app is shared in this case")
+  public void pluginWithDependencyAndConflictingVersionSharedByApp() throws Exception {
+    final ApplicationFileBuilder differentLibPluginAppFileBuilder = appFileBuilder("appWithLibDifferentThanPlugin")
+        .definedBy("app-plugin-different-lib-config.xml")
+        .dependingOn(echoPluginWithLib1)
+        .dependingOnSharedLibrary(new JarFileBuilder("barUtils2_0", barUtils2_0JarFile))
+        .containingClass(pluginEcho2TestClassFile, "org/foo/echo/Plugin2Echo.class");
 
-    ApplicationFileBuilder sharedLibPluginAppFileBuilder = appFileBuilder("shared-plugin-lib-app")
-        .definedBy("app-with-echo1-plugin-config.xml").dependingOn(echoPluginWithoutLib1)
-        .dependingOnSharedLibrary(new JarFileBuilder("barUtils", barUtils1_0JarFile));
+    addPackedAppFromBuilder(differentLibPluginAppFileBuilder);
 
-    addPackedAppFromBuilder(sharedLibPluginAppFileBuilder);
+    startDeployment();
+
+    assertDeploymentSuccess(applicationDeploymentListener, differentLibPluginAppFileBuilder.getId());
+
+    executeApplicationFlow("main");
+  }
+
+  @Test
+  @Description("Ensure that when a plugin2 depends on plugin1, when using something exported from plugin1, that is used")
+  public void pluginDependingAndExportingFromOtherPlugin() throws Exception {
```

Comment:
Add a test to ensure that, when an artifact contains package P1, which is also exported on the region, and that P1 is part on the artifact exported API, then the artifact uses P1 form the region.